go/tasks/plugins/k8s/ray/config.go

```diff
@@ -0,0 +1,41 @@
+package ray
+
+import (
+	pluginsConfig "github.com/flyteorg/flyteplugins/go/tasks/config"
+)
+
+//go:generate pflags Config --default-var=defaultConfig
+
+var (
+	defaultConfig = Config{
+		ShutdownAfterJobFinishes: true,
+		TTLSecondsAfterFinished:  3600,
+		ServiceType:              "NodePort",
+	}
+
+	configSection = pluginsConfig.MustRegisterSubSection("ray", &defaultConfig)
+)
+
+// Config is config for 'ray' plugin
+type Config struct {
+	// ShutdownAfterJobFinishes will determine whether to delete the ray cluster once rayJob succeed or failed
+	ShutdownAfterJobFinishes bool `json:"shutdownAfterJobFinishes,omitempty"`
+
+	// TTLSecondsAfterFinished is the TTL to clean up RayCluster.
+	// It's only working when ShutdownAfterJobFinishes set to true.
+	TTLSecondsAfterFinished int32 `json:"ttlSecondsAfterFinished,omitempty"`
+
+	// Kubernetes Service Type, valid values are 'ClusterIP', 'NodePort' and 'LoadBalancer'
+	ServiceType string `json:"serviceType,omitempty"`
+
+	// ClusterSelector is used to select running rayclusters by labels
+	ClusterSelector map[string]string `json:"clusterSelector,omitempty"`
```

Comment:
I'm not sure I fully understand what this configuration option does. Is this a FlytePropeller level option? Or is it better set on each individual Ray task?

Reply:
[ClusterSelector](https://github.com/ray-project/kuberay/blob/0c98cf2797e6e0e96fbc9889509f622373f7d89f/ray-operator/apis/ray/v1alpha1/rayjob_types.go#L52-L53) is used by ray-operator. If the ClusterSelector is set, ray operator will submit the ray job to that cluster.
I assume users only have one cluster, so all the tasks will run on that cluster. Thus, I add it to the propeller config.
We could also add `ClusterSelector` to the flytekit config in the future if we have two tasks and these tasks run on different clusters. Then, the value in the flytekit config will override the one in propeller config.